contracts/chain-adapters/Solana_Adapter.sol

```diff
@@ -0,0 +1,193 @@
+        }
+
+        // TODO: consider if we need also to emit the translated message.
+        emit MessageRelayed(target, message);
+    }
+
+    /**
+     * @notice Bridge tokens to Solana.
+     * @dev Only allows bridging USDC to Solana spoke pool.
+     * @param l1Token L1 token to deposit.
+     * @param l2Token L2 token to receive.
+     * @param amount Amount of L1 tokens to deposit and L2 tokens to receive.
+     * @param to Bridge recipient.
+     */
+    function relayTokens(
+        address l1Token,
+        address l2Token,
+        uint256 amount,
+        address to
+    ) external payable override {
+        if (l1Token != address(usdcToken)) {
+            revert InvalidL1Token(l1Token);
+        }
+        if (l2Token != SOLANA_USDC_ADDRESS) {
+            revert InvalidL2Token(l2Token);
+        }
+        if (amount > type(uint64).max) {
+            revert InvalidAmount(amount);
+        }
+        if (to != SOLANA_SPOKE_POOL_ADDRESS) {
+            revert InvalidTokenRecipient(to);
+        }
+
+        _transferUsdc(SOLANA_SPOKE_POOL_USDC_VAULT, amount);
+
+        // TODO: consider if we need also to emit the translated addresses.
+        emit TokensRelayed(l1Token, l2Token, amount, to);
+    }
+
+    /**
+     * @notice Helper to map a Solana address to an Ethereum address representation.
+     * @dev The Ethereum address is derived from the Solana address by truncating it to its lowest 20 bytes. This same
+     * conversion must be done by the HubPool owner when adding Solana spoke pool and setting the corresponding pool
+     * rebalance and deposit routes.
+     * @param solanaAddress Solana address (Base58 decoded to bytes32) to map to its Ethereum address representation.
+     * @return Ethereum address representation of the Solana address.
+     */
+    function _trimSolanaAddress(bytes32 solanaAddress) internal pure returns (address) {
+        return address(uint160(uint256(solanaAddress)));
+    }
+
+    /**
+     * @notice Translates a message to enable/disable a route on Solana spoke pool.
+     * @param message Message to translate, expecting setEnableRoute(address,uint256,bool).
+     * @return Translated message, using setEnableRoute(bytes32,uint64,bool).
+     */
+    function _translateSetEnableRoute(bytes calldata message) internal view returns (bytes memory) {
+        (address originToken, uint256 destinationChainId, bool enable) = abi.decode(
+            message[4:],
+            (address, uint256, bool)
+        );
+
+        if (originToken != SOLANA_USDC_ADDRESS) {
+            revert InvalidOriginToken(originToken);
+        }
+
+        if (destinationChainId > type(uint64).max) {
+            revert InvalidDestinationChainId(destinationChainId);
+        }
+
+        return
+            abi.encodeWithSignature(
+                "setEnableRoute(bytes32,uint64,bool)",
+                SOLANA_USDC_BYTES32,
+                uint64(destinationChainId),
+                enable
+            );
```

Comment:
OOC, why not do this mapping on the Solana side?

In theory, could we not have a mapping (from truncated address to full address) exist on the Solana side that could be modified by an admin call. Then, the flow to add a new token would require no contract changes:
1. Send admin call to update the mapping with a new element.
2. Add the route to use the newly added element.

Side note: this is somewhat similar to something we have to do on [arbitrum](https://github.com/across-protocol/contracts/blob/98c761ee9ca3a496b4a368bdf12743d727644138/contracts/Arbitrum_SpokePool.sol#L80-L82).